functorch/_src/eager_transforms.py

```diff
@@ -854,11 +884,26 @@ def wrapper_fn(*args):
         basis = tree_unflatten(flat_basis, primals_spec)
 
         def push_jvp(basis):
-            _, jvp_out = jvp(f_wrapper, primals, basis)
+            output = jvp(f_wrapper, primals, basis, has_aux=has_aux)
+            if has_aux:
+                _, jvp_out, aux = output
+                return jvp_out, aux
+            _, jvp_out = output
             return jvp_out
 
         results = vmap(push_jvp)(basis)
+        if has_aux:
+            results, aux = results
+            # aux is in the standard basis format, e.g. NxN matrix
+            # We need to fetch the first element as original `func` output
+            flat_aux, aux_spec = tree_flatten(aux)
+            flat_aux = [value[0] for value in flat_aux]
+            aux = tree_unflatten(flat_aux, aux_spec)
```

Comment:
Hmm, I don't follow -- what's going on here? Do you have an example?

Reply:
Oh, I get it now... we didn't actually want to vmap over the aux computation, we want to return the actual aux output of f(*primals) to the user...